### PR TITLE
[FIX] #1238 howto search cleaning after navigation

### LINKED ISSUE #1238: [bug] - howto search persists after navigation
**Describe the bug**
When using the search box to find/filter howtos the sort order remains in the howto store regardless of any future navigation events. This is a bit confusing because the search param in the URL does disappear.

**To Reproduce**

1. Go to howto page
2. Enter text in search box
3. Follow howto link
4. Click howto in main menu bar to go to howto home page

**Expected behavior**
When the howto search param is cleared from the URL the howto page (e.g. following navigation), the howto list should also revert to default list. Similarly, if a user presses the back button on their browser to return to the search page both browser URL and search order should be preserved/displayed (currently correct)

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by jparadasb**
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

I added a flush of the search value in the howToStore when the component invokes `componentWillUnmount` and the respective e2e in cypress test.

## Git Issues

Closes #1238 

## Screenshots/Videos

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on slack in the `platform-dev` channel.

### CODE REVIEW

**Comment by thisislawatts** [comment]
@jparadasb did you see any of these e2e tests failures locally?

**Comment by jparadasb** [comment]
No, indeed, all those tests passed locally. It's weird 😟
Screen Shot 2021-12-28 at 23 27 54

**Comment by thisislawatts** [comment]
Thanks for confirming @jparadasb, I also pulled the branch locally and got it working 🤔 I will poke around a bit more and see what I can find.

File: src/pages/Howto/Content/HowtoList/HowtoList.tsx [hunk 1]
```diff
@@ -85,6 +85,10 @@ export class HowtoList extends React.Component<any, IState> {
     }
   }
 
+  componentWillUnmount(): void {
+    this.props.howtoStore.updateSearchValue('')
+  }
```
Comment (thisislawatts) [inline src/pages/Howto/Content/HowtoList/HowtoList.tsx:90]: One little line, how is it resulting in so much instability in our tests **only** within the CI context. 🤔

**Comment by thisislawatts** [comment]
Okay okay, apologies @jparadasb the error is non obvious but I believe it is entirely our fault. It is tied the way we've configured our CI pipeline, as the e2e step consumes secrets. By default these are **not** passed to forked versions of the repo. 

> a reviewer decided it (a PR) was safe to run, they could kick off the trusted job by pushing the PR’s commits to a branch of the main repository. 

I pulled the branch from your repo and pushed it to the primary repo, as detailed in this blog post; [Triggering trusted CI jobs on untrusted forks](https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/)

Further reading:
* https://circleci.com/docs/2.0/oss/#build-pull-requests-from-forked-repositories

**Review by thisislawatts — APPROVED** [review]

**Comment by jparadasb** [comment]
Oh! Good catch!! 🙌

**Pull request merged**